question.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const inquirer = require('inquirer')
+const path = require('path')
+
+module.exports = function question (dir) {
+  return inquirer.prompt([
+    {
+      type: 'input',
+      name: 'name',
+      message: 'package name',
+      default: function () {
+        return path.basename(dir)
+      },
+      validate: function (input) {
+        if (/^([A-Za-z-_\d])+$/.test(input)) {
+          return true
+        } else {
+          return 'Project name may only include letters, numbers, underscores and hashes.'
+        }
+      }
+    },
+    {
+      type: 'input',
+      name: 'version',
+      message: 'version',
+      validate: function (input) {
+        if (/[0-9]+\.[0-9]+\.[0-9]+/.test(input)) {
```

Comment:
I think we should be using the semver module to validate this.

Reply:
Sure will patch in once github status is green